### DISCUSSION TITLE
How to display all features in the summary plot ?

### DISCUSSION THREAD

**Discussion by saneehaAmir** [shap/shap#3119] [General] [answered]
Hello,

I am working with 512 features and i need to show see the bar summary plot for all features. However the plot only shows 20 features. I have also tried max_display=none option but it is not working. 

The code is :
explainer = shap.Explainer(clf)
shap_values = explainer.shap_values(X_test)
shap.summary_plot(shap_values, X_test,max_dispaly=None)

Your help is appreciated

**Comment by thatlittleboy** [comment]
What does "not working" mean exactly?  Still only 20 features are shown, or..?

**Reply by saneehaAmir** [reply]
Yes only 20 features are shown still

**Reply by thatlittleboy** [reply, marked as answer]
Can you try using `shap.plots.beeswarm` instead of `summary_plot`?

```python
explainer = shap.Explainer(clf)
explanation = explainer(X_test)
shap.plots.beeswarm(explanation, max_display=None)
```

Please consult the API notebooks if still in doubt: https://shap.readthedocs.io/en/latest/example_notebooks/api_examples/plots/beeswarm.html

**Reply by saneehaAmir** [reply]
I have tried beeswarm plot. It works fine for 2 class problem and displays full feature list. However it raises the following error with 4 class problem.

ValueError: The beeswarm plot does not support plotting explanations with instances that have more than one dimension!

Much thanks for your help
